### PR TITLE
[FLINK-27317][build] Don't fork build for attaching sources 

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1224,12 +1224,12 @@ under the License.
 					<plugin>
 						<groupId>org.apache.maven.plugins</groupId>
 						<artifactId>maven-source-plugin</artifactId>
-						<version>2.2.1</version><!--$NO-MVN-MAN-VER$-->
```
Comment (dmvk) [inline pom.xml:1227]: is there any particular reason for the upgrade?

**Review by dmvk — COMMENTED** [review]
This is some dark working directory magic. :) How about making the `.scalafmt.conf` path relative to the base directory instead? Are there any other consequences of using `-no-fork`?

eg.
```diff --git a/flink-scala/pom.xml b/flink-scala/pom.xml
index ff31c8996bb..ad805f54d6d 100644
--- a/flink-scala/pom.xml
+++ b/flink-scala/pom.xml
@@ -146,7 +146,7 @@ under the License.
 					<scala>
 						<scalafmt>
 							<version>${spotless.scalafmt.version}</version>
-							<file>../.scalafmt.conf</file>
+							<file>${project.basedir}/../.scalafmt.conf</file>
 						</scalafmt>
 						<licenseHeader>
 							<content>${spotless.license.header}</content>
```

**Comment by zentol** [comment]
Using `${project.basedir}` is a good idea. I'm not aware of any downsides in using `-no-fork`.

File: pom.xml (hunk 1 above)
Comment (zentol) [reply pom.xml:1227]: No; it just felt like a good opportunity to do so since we're looking at the plugin anyway.

**Comment by zentol** [comment]
I would use the `-no-fork` goal though in any case because it saves some time.

**Review by dmvk — APPROVED** [review]
LGTM, thanks for the fix @zentol! Good job.

**Comment by zentol** [comment]
@flinkbot run azure

**Comment by zentol** [comment]
@flinkbot run azure

**Pull request merged**